common/maven-url-handler/src/main/java/io/helidon/build/maven/url/MavenURLHandler.java

```diff
@@ -0,0 +1,464 @@
+                JarFile jarFile = new JarFile(jar);
+                Enumeration<JarEntry> entries = jarFile.entries();
+                while (entries.hasMoreElements()) {
+                    JarEntry entry = entries.nextElement();
+                    File out = getFileFromArchive(path, outDirectory, jarFile, entry);
+                    if (out != null) {
+                        return out;
+                    }
+                }
+                throw new IOException(String.format("File %s was not found into jar : %s.", path, jarName));
+            }
+
+            /**
+             * Visit a zip file.
+             *
+             * @param zipName       Jar file name.
+             * @param path          path to a file into the jar file.
+             * @return              the file targeted by the path.
+             * @throws IOException  if file is not present.
+             */
+            public File visitZip(String zipName, String path) throws IOException {
+                File zip = visit(zipName);
+                File outDirectory = new File(zip.getParent());
+                ZipFile zipFile = new JarFile(zip);
+                Enumeration<? extends ZipEntry> entries = zipFile.entries();
+                while (entries.hasMoreElements()) {
+                    ZipEntry entry = entries.nextElement();
+                    File out = getFileFromArchive(path, outDirectory, zipFile, entry);
+                    if (out != null) {
+                        return out;
+                    }
+                }
+                throw new IOException(String.format("File %s was not found into zip : %s.", path, zipName));
+            }
+
+            private File getFileFromArchive(String path, File directory, ZipFile zipFile, ZipEntry entry) throws IOException {
+                String name = entry.getName();
+                if (name.equals(path)) {
+                    InputStream is = zipFile.getInputStream(entry);
+                    File out = createFile(directory, name);
+                    FileOutputStream fos = new FileOutputStream(out);
+                    int c;
+                    while ((c = is.read()) != -1) {
+                        fos.write(c);
+                    }
+                    fos.close();
+                    is.close();
+                    return out;
+                }
+                return null;
+            }
+
+            private File createFile(File directory, String name) throws IOException {
+                Files.createDirectories(directory.toPath().resolve(name).getParent());
+                Files.write(directory.toPath().resolve(name), new byte[0], StandardOpenOption.CREATE);
+                return new File(directory.toPath().resolve(name).toString());
+            }
+
+            private File[] resolveFiles(File file) {
+                return file.listFiles();
+            }
+        }
+
+    }
+
+    static class Parser {
```

Comment:
If you decide to externalize these inner classes, this one could be called MavenURLParser.